### PR TITLE
Change jx_iterate_keys to return a const char *

### PR DESCRIPTION
**Pull request by trshaffer**
From the email discussion started by @btovar

### CODE REVIEW

**Comment by dthain** [comment]
@btovar What do you think?

**Comment by btovar** [comment]
I have a couple on questions in:
https://github.com/cooperative-computing-lab/cctools/pull/1761/files

**Comment by dthain** [comment]
Is this mergeable?

File: dttools/src/jx.c [hunk 1]
```diff
@@ -667,21 +667,30 @@ struct jx * jx_iterate_array(struct jx *j, void **i) {
...
+	if (!(*i)) {
 		if (!jx_istype(j, JX_OBJECT)) return NULL;
-		*i = j->u.pairs;
-		return *i ? ((struct jx_pair *) *i)->key : NULL;
+		*i = &j->u.pairs;
+	}
+
+	struct jx_pair **p = *i;
+	while (*p) {
+		if (jx_istype((*p)->value, JX_STRING)) break;
```
[the remark this answers is not in the document]
Comment (trshaffer) [reply dttools/src/jx.c]: So keys *should* always be strings. But it's possible to put a JX expression there, or just make a non-string key, so I don't want to segfault if that assumption doesn't hold. We wouldn't need to be as careful if there were a type-level distinction between JSON and JX.

**Pull request merged**